### ISSUE TITLE
Allow Building xml scenes using xacro

### ISSUE THREAD

**Issue by perezsolerj** [uji-ros-pkg/underwater_simulation#36]
Scenes files are getting too big to mantain and most of the blocks are copy-pasted to other scenes making difficult to achieve consistency. We could use xacro http://wiki.ros.org/xacro to compose scenes at build time from functional blocks such as girona500 vehicle, stereo camera, multibeam or black box. 

This way we could create configuration blocks for sensors, vehicles etc easier to mantain and change.

**Comment by miquelmassot** [comment]
Today I started playing with xacro for better consistency. I will be adding new macros and when it's finished I'll PR you.
Take a look: https://github.com/srv/underwater_simulation/commit/78fab135a34c47989e9decb03b71cf19f925befc

**Comment by perezsolerj** [comment]
Finally I made some work regarding this, I created 4 libraries:

Common: Has basic generic macros for UWSimScenes
DeviceLibrary: intended for sensors and interfaces, I added some as example.
ObjectLibrary: Common used objects such as cirs terrain or blackbox should be available here, so there is no need to remember 3d mesh file, offsets or physics config to add them on a scene.
VehicleLibrary: library for vehicles (uses deviceLibrary to add sensors). This way we can easily maintain different configs for vehicles and add the whole vehicle in a scene instead of copying it.

xacroExample.xml.xacro is an example of use, as can be seen only a few lines are needed to create a scene similar to cirs.xml .

Finally I added some lines to CMakeLists so .xml.xacro files present on data/scenes are automatically built on catkin_make. Also, if any of this files suffers any changes is rebuilt on the next catkin_mak, so there is no need to call "rosrun xacro....".

The only drawback is I had to add an attribute to the DTD so it does not complain on xacro generated scenes.

More details on commit cd41a66912e2faf3c55099cc4858429c2c99304a
